src/compiler/binder.ts

```diff
@@ -2457,13 +2456,8 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
             ) {
                 file.bindDiagnostics.push(createDiagnosticForNode(node, getStrictModeIdentifierMessage(node), declarationNameToString(node)));
             }
-            else if (originalKeywordKind === SyntaxKind.AwaitKeyword) {
-                if (isExternalModule(file) && isInTopLevelContext(node)) {
-                    file.bindDiagnostics.push(createDiagnosticForNode(node, Diagnostics.Identifier_expected_0_is_a_reserved_word_at_the_top_level_of_a_module, declarationNameToString(node)));
-                }
-                else if (node.flags & NodeFlags.AwaitContext) {
-                    file.bindDiagnostics.push(createDiagnosticForNode(node, Diagnostics.Identifier_expected_0_is_a_reserved_word_that_cannot_be_used_here, declarationNameToString(node)));
-                }
+            else if (originalKeywordKind === SyntaxKind.AwaitKeyword && (node.flags & NodeFlags.AwaitContext || isExternalModule(file))) {

```

Comment:
Perhaps this would deserve a new diagnostic. The other one (with `node.flags & NodeFlags.AwaitContext`) didn't have a specific diagnostic meant for it so I skipped adding one here as well. 

I think though that it could be nice to add a specific diagnostic here (and maybe add a specific one for that other case as well while at it). If you agree - should this be a complete new diagnostic code or could the existing code be reused? Probably a new one makes more sense.

Reply:
I believe technically this is allowed in non-strict CJS, but I'd rather not make weird exceptions.

Reply:
yes, it only technically has to be disallowed in modules 😉 Isn't allowing it in TS files with script targets a weird exception already? 😅

Reply:
It's not allowed in non-strict CJS, it's **always** allowed there. The actual rule is that this is disallowed in modules - not that it's disallowed in strict mode 🤷